modules/db/src/main/java/org/mule/module/db/internal/domain/executor/StoredProcedureExecutor.java

```diff
@@ -40,7 +40,25 @@ protected Object doExecuteQuery(DbConnection connection, Statement statement, Qu
     @Override
     protected Object doExecuteQuery(DbConnection connection, Statement statement, Query query, AutoGeneratedKeyStrategy autoGeneratedKeyStrategy) throws SQLException
     {
-        autoGeneratedKeyStrategy.execute(statement, query.getQueryTemplate());
+        try
+        {
+            autoGeneratedKeyStrategy.execute(statement, query.getQueryTemplate());
```

Comment:
Is it an idea to implement this using a Decorator/Wrapper pattern around a AutoGeneratedKeyStrategy. You could instantiate this Decorator/Wrapper inside StoredProcedureMessageProcessor.